### DISCUSSION TITLE
Is there a way to only execute the command one time when you hold the keybind?

### DISCUSSION THREAD

**Discussion by zeozeozeo** [ajaxorg/ace#4916] [Q&A] [answered]
I have `Ctrl+S` keybind which saves the file currently opened in the editor. When I hold the keybind, it starts repeatedly saving the file. Is there a way for it to not do that?
This is the code I'm using:
```javascript
editor.commands.addCommand({
    name: "save",
    bindKey: { win: "Ctrl-S", mac: "Cmd-S" },
    exec: (editor) => {
        saveFile();
    },
});
```

**Comment by nightwing** [comment, marked as answer]
Ace does not have a way to register commands only for key down, but one workaround is to use 
```
if (window.event && window.event.repeat) return
```
before calling saveFile, 
That said, i think most editors simply use save state to prevent repeated calls and don't distinguish between first and repeated calls.
